packages/maskbook/src/components/RequestPermission/RequestPermission.tsx

```diff
@@ -20,13 +22,14 @@ const useStyles = makeStyles((theme: Theme) => ({
 interface RequestPermissionProps {
     permission: browser.permissions.Permissions
     onRequestApprove(): void
-    onCancel(): void
+    onReject(): void
+    reason: string
 }
 export function RequestPermission(props: RequestPermissionProps) {
     const classes = useStyles()
     return (
         <Card className={classes.root}>
-            <DialogTitle>Mask needs the following permissions</DialogTitle>
+            <DialogTitle>{props.reason || 'Maskbook needs the following permissions'}</DialogTitle>
             <DialogContent>
```

Comment:
It's possible that the DialogContent can be removed because the permissions and origins we are requesting will be prompted and shown to the user by the browser itself anyways. Instead we could describe the reason "why" we need this permission here and what we're trying to do, and let the browser ask for specific permissions. That's why I added the `reason` prop here